frontend/src/components/VisaStatusOverviewList/OverviewContent.js

```diff
@@ -0,0 +1,243 @@
+import React, { useMemo, useState } from "react";
+import PropTypes from "prop-types";
+import { useSelector } from "react-redux";
+import { Row, Col, Button, Tooltip, Space, Collapse, Tag, Modal } from "antd";
+import { MailOutlined, QqOutlined, EllipsisOutlined, PlusOutlined } from "@ant-design/icons";
+import { useTranslation } from "react-i18next";
+import { useScreenXS } from "../../hooks";
+import { makeNewestVisaStatusSelector } from "../../redux/selectors";
+import { findEmbassyAttributeByCode } from "../../utils/USEmbassy";
+import { overviewAttrProps, newestOverviewProps, overviewProps } from "./overviewPropTypes";
+import "./VisaStatusOverview.less";
+
+const { Panel } = Collapse;
+
+const QQTGSubsModalHook = (qq, tg) => {
+    const { t } = useTranslation();
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    return [
+        <Modal
+            title={t("QQTGModalTitle")}
+            visible={isModalVisible}
+            onOk={() => setIsModalVisible(false)}
+            onCancel={() => setIsModalVisible(false)}
+        >
+            <p>
+                {t("QQTGModalContentTG")}
+                <a href={tg} target="_blank" rel="noreferrer">
+                    {tg}
+                </a>
+            </p>
+            <p>{t("QQTGModalContentQQDesc")}</p>
+            {qq.map((content, index) => (
+                <p>
+                    {t("QQTGModalContentQQ", { index: index + 1 })}
+                    {content}
+                </p>
+            ))}
+        </Modal>,
+        setIsModalVisible,
+    ];
+};
```

Comment:
A custom hook should _not_ return a React component. The right thing to do is to define a new component `QQTGSubsModal` and pass down the pop up control

```js
const QQTGSubsModal = ({ qqGroups, tgLink, modalControl }) => {
    const hook = useHook(); // use the hook here
    //...

    return (
        <Modal onOk={() => modalControl()} onCancel={() => modalControl()}>
            <OtherContent />
        </Modal>
    );
}
```